lua/packer/compile.lua

```diff
@@ -164,19 +164,43 @@ local function timed_chunk(chunk, name, output_table)
   return output_table
 end
 
+local timed_install_check_fmt = [=[time([[%s]], true)
+vim.loop.fs_lstat([[%s]], function(err, stat)
+  if err == nil and stat ~= nil then
+    vim.schedule(function()
```

Comment:
@wbthomason is it worth checking how fast the synchronous version of this is before using the async version to avoid the flickering issues and other bugs that are likely to crop up with this running asynchronously. I think `fs_stat` is relatively quick, alternatively maybe read the contents of the start and opt directory once and get a list of directories and just check the plugin name is in a list of dirs within either dir. That way, the existence check only happens once by just reusing the list of directories that were in `start/opt` dirs?

Reply:
> is it worth checking how fast the synchronous version of this is before using the async version to avoid the flickering issues and other bugs that are likely to crop up with this running asynchronously

Probably, yeah. I was basing the decision to do async from the start on the cost of scanning for installed plugins that I saw in #428, but if async causes flickering on every startup, it's worth testing the sync alternative.

> alternatively maybe read the contents of the start and opt directory once and get a list of directories and just check the plugin name is in a list of dirs within either dir

This is more expensive than I'd like for something that runs at every start, ideally. That's why e.g. we defer checking for installed plugins until a plugin management operation happens already.